### PR TITLE
TMDM-13179 [Postgres] It's not possible to activate SSL communication between MDM Server and Database

### PR DESCRIPTION
**Pull request by hw-zhu**
https://jira.talendforge.org/browse/TMDM-13179
**What is the current behavior?** (You should also link to an open issue here)
Cannot support any parameter in the connection url, If the URL already contains a parameter then new parameter should not be append by a '?' but by a &.

**What is the new behavior?**
Using new method to create Connection object, that the new Properties argument can be used to pass arbitrary string tag/value pairs as connection arguments.Normally at least "user" and "password" properties should be included in the Properties object.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/datasource/RDBMSDataSource.java [hunk 1]
```diff
@@ -421,6 +424,65 @@ public Properties getAdvancedPropertiesIncludeUserInfo() {
...
+        Driver driver = createDriver(dataSource);
+        Properties properties = new Properties();
+        properties.put("user", dataSource.getUserName()); //$NON-NLS-1$
+        properties.put("password", dataSource.getPassword()); //$NON-NLS-1$
+
+        String connectionURL = dataSource.getConnectionURL();
+        return driver.connect(connectionURL, properties);
+    }
+
+    private static String getConnectionURL(RDBMSDataSource dataSource) {
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/datasource/RDBMSDataSource.java]: I remembered we  already changed the function name to `getInitConnnectionURL`, right?

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/datasource/RDBMSDataSource.java [hunk 2]
```diff
@@ -421,6 +424,65 @@ public Properties getAdvancedPropertiesIncludeUserInfo() {
...
+     * Connecting to special DB Using JDBC Driver,need to do the following steps:
+     * <ul><li>
+     * 1. Load the DB Connector/J into your program.
+     * </li><li>
+     * 2. Create a new Connection object from the Driver class. Then you can use this Connection object to execute queries.
+     * </li></ul>
+     * @param dataSource
+     * @return
+     */
+    public static Connection getConnection(RDBMSDataSource dataSource) throws SQLException {
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/datasource/RDBMSDataSource.java]: And this function name would be `getInitedConnection`, right?

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/prepare/PostgresStorageCleaner.java [hunk 3]
```diff
@@ -44,8 +43,7 @@ public void clean(Storage storage) {
                 throw new IllegalArgumentException("Data source '" + dataSource.getName() + "' does not define initialization information.");
             }
 
-            Driver driver = (Driver) Class.forName(dataSource.getDriverClassName()).newInstance();
-            Connection connection = driver.connect(dataSource.getInitConnectionURL() + "?user=" + dataSource.getInitUserName() + "&password=" + dataSource.getInitPassword(), new Properties());  //$NON-NLS-1$ //$NON-NLS-2$
+            Connection connection = RDBMSDataSource.getConnection(dataSource);
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/prepare/PostgresStorageCleaner.java]: Would it be `getInitConnection`?

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/datasource/RDBMSDataSource.java [hunk 4]
```diff
@@ -421,6 +424,65 @@ public Properties getAdvancedPropertiesIncludeUserInfo() {
         return properties;
     }
 
+    private static Driver createDriver(RDBMSDataSource dataSource) throws SQLException {
+        try {
+            return (Driver) Class.forName(dataSource.getDriverClassName()).newInstance();
+        } catch (Exception e) {
+            throw new RuntimeException("Exception occurred during initialization of database", e);
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/datasource/RDBMSDataSource.java]: "Exception occurred during initialization of database."

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/datasource/RDBMSDataSource.java [hunk 5]
```diff
@@ -421,6 +424,65 @@ public Properties getAdvancedPropertiesIncludeUserInfo() {
...
+        }
+    }
+
+    /**
+     * Connecting to special DB Using JDBC Driver,need to do the following steps:
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/datasource/RDBMSDataSource.java]: `Connecting to special DB Using JDBC Driver, need to do the following steps`

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/datasource/RDBMSDataSource.java [hunk 6]
```diff
@@ -421,6 +424,65 @@ public Properties getAdvancedPropertiesIncludeUserInfo() {
...
+        Properties properties = new Properties();
+        properties.put("user", dataSource.getInitUserName()); //$NON-NLS-1$
+        properties.put("password", dataSource.getInitPassword()); //$NON-NLS-1$
+
+        String connectionURL = getInitConnectionURL(dataSource);
+        return driver.connect(connectionURL, properties);
+    }
+
+    /**
+     * Connecting to special DB Using JDBC Driver,need to do the following steps:
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/datasource/RDBMSDataSource.java]: `Connecting to special DB Using JDBC Driver, need to do the following steps:`

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/datasource/RDBMSDataSource.java [hunk 7]
```diff
@@ -421,6 +424,65 @@ public Properties getAdvancedPropertiesIncludeUserInfo() {
...
+     * <ul><li>
+     * 1. Load the DB Connector/J into your program.
+     * </li><li>
+     * 2. Create a new Connection object from the Driver class. Then you can use this Connection object to execute queries.
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/datasource/RDBMSDataSource.java]: `Get an existing Connection object from the Driver class. Then you can use this Connection object to execute queries.`

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/datasource/RDBMSDataSource.java [hunk 8]
```diff
@@ -421,6 +424,65 @@ public Properties getAdvancedPropertiesIncludeUserInfo() {
...
+    private static Driver createDriver(RDBMSDataSource dataSource) throws SQLException {
+        try {
+            return (Driver) Class.forName(dataSource.getDriverClassName()).newInstance();
+        } catch (Exception e) {
+            throw new RuntimeException("Exception occurred during initialization of database", e);
+        }
+    }
+
+    /**
+     * Create a JDBC Connecting without assign database, need to do the following steps
```
Comment (lpw82gf) [inline org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/datasource/RDBMSDataSource.java]: `Create a JDBC Connection`

**Pull request merged**